### PR TITLE
760 add version endpoint appconfig

### LINKED ISSUE #760: [FEATURE] Add /version endpoint to AppConfig Building Block
**Is your feature request related to a problem? Please describe.**
Currently, we don't have a way to retrieve the Building Block's version number through API.

**Describe the solution you'd like**
We need to add a /version endpoint that returns the current version number. OpenAPI specification YAML file needs to be updated with this endpoint. Basically, the building block needs to capture its docker image tag and return that in the response. Response Content-Type can be text/plain. If reading the docker image tag from within the container is not straightforward, the building block can read the version number in the YAML file.

**Additional context**
Epic: #738

### PR DESCRIPTION
**Pull request by minump**
## Description
Add version endpoint for AppConfig

**Fixes #760**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: appconfigservice/appconfig.yaml [hunk 1]
```diff
@@ -131,6 +131,21 @@ paths:
...
         500:
           description: Internal error
+  /app/configs/version:
+    get:
+      tags:
+        - App Configuration
+      summary: version endpoint
+      description: Public endpoint which returns the latest version of appconfig BB
+      responses:
+        200:
```
Comment (sandeep-ps) [inline appconfigservice/appconfig.yaml:141]: It looks like by default the response content type is application/json. As per the issue description, we need to have content type text/plain for this end point. Please see https://swagger.io/docs/specification/describing-responses/ for reference and fix this. Thanks.

File: appconfigservice/api/controllers/app.py [hunk 2]
```diff
@@ -43,6 +44,17 @@ def configs_ok_search():
     return msg
 
 
+def configs_version_search():
+    """
+    Method to return AppConfig BB version number
+    Reads yaml file and returns version number
+    Return : plain text - version number
+    """
+    appconfig_yaml = open("appconfig.yaml")
```
Comment (sandeep-ps) [inline appconfigservice/api/controllers/app.py:53]: @minump, I just realized why this was working when running using PyCharm. With PyCharm, I think it looks at the repository and tries to find this file. 

Please run the container using Docker. There is an error that it's not able to find the file. 

```
File "/app/appconfigservice/api/controllers/app.py", line 53, in configs_version_search
    appconfig_yaml = open("appconfig.yaml")
FileNotFoundError: [Errno 2] No such file or directory: 'appconfig.yaml'
```
I think this goes back to why it's important to run modules using Docker when reviewing code (note to myself). Please fix this. Thanks.

Comment (robkooper) [reply appconfigservice/api/controllers/app.py:53]: one option to look at is:
```
from flask import current_app

def function():
  with app.app.app_context():
    current_app.config['env'] = 'dev'
```

see https://github.com/zalando/connexion/issues/119

Comment (sandeep-ps) [reply appconfigservice/api/controllers/app.py]: Hi, Rob,

I probably did not understand what you exactly meant. Could you please share a little more details? Just to clarify, this module is called App Config, and this PR was not about this module's config.

**Comment by sandeep-ps** [comment]
For this PR, you could update the Dockerfile to copy the YAML file to the appropriate folder so that you can use open('appconfig.yaml) file when running 'python api/app_config_...'. This way, it'll work when both running as Docker and directly using Python. I'm not getting other ideas at this time. I checked if we can get yaml file data from connexion Flask object, but with no luck. Please let me know if you have any questions. Thanks.

**Comment by minump** [comment]
updated PR with necessary changes for dockerfile

**Review by sandeep-ps — APPROVED** [review]

**Comment by sandeep-ps** [comment]
Tested the endpoint using both Docker and running directly using Python. Both are working correctly.

**Pull request merged**